src/main/java/software/amazon/msk/auth/iam/internals/MSKCredentialProvider.java

```diff
@@ -136,13 +139,20 @@ EnhancedProfileCredentialsProvider createEnhancedProfileCredentialsProvider(Stri
                 }
                 String sessionName = Optional.ofNullable((String) optionsMap.get(AWS_ROLE_SESSION_KEY))
                         .orElse("aws-msk-iam-auth");
-                return createSTSRoleCredentialProvider((String) p, sessionName);
+                String stsRegion = Optional.ofNullable((String)optionsMap.get(AWS_STS_REGION))
+                        .orElse("aws-global");
+                return createSTSRoleCredentialProvider((String) p, sessionName, stsRegion);
             });
         }
 
         STSAssumeRoleSessionCredentialsProvider createSTSRoleCredentialProvider(String roleArn,
-                String sessionName) {
-            return new STSAssumeRoleSessionCredentialsProvider.Builder(roleArn, sessionName).build();
+                                                                                String sessionName, String stsRegion) {
+            AWSSecurityTokenService stsClient = AWSSecurityTokenServiceClientBuilder.standard()
+                    .withRegion(stsRegion)
```

Comment:
I tried out the PR on a test setup with the awsStsRegion set and it kept using the global `sts.amazonaws.com`. It did not work, until I also set the regional endpoint for sts. The [documentation](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_temp_enable-regions.html) also seems to indicate that. Could we change to accept a awsStsEndpoint argument in addition to awsStsRegion ?

Code snippet I tried below:
``
// The stsEndpoint is of the form: "sts.us-west-2.amazonaws.com".
 AwsClientBuilder.EndpointConfiguration endpointConfiguration = new AwsClientBuilder.EndpointConfiguration(stsEndpoint, stsRegion); 
AWSSecurityTokenService stsClient = AWSSecurityTokenServiceClientBuilder.standard()
                    .withEndpointConfiguration(endpointConfiguration)
                    .build();
``

Reply:
This was the first line of investigation, but I found in the [SDK docs](https://docs.aws.amazon.com/sdkref/latest/guide/setting-global-sts_regional_endpoints.html) that setting the region along side `AWS_STS_REGIONAL_ENDPOINTS=regional` environment variable leads to the same end result. 

I believe if we provide the endpoint directly then the region becomes redundant (as the endpoint will be used instead). What do you feel it's be best way to solve this? I'm happy to make any changes to the PR.